File: AndroidBasics/notes/src/main/java/pan/alexander/notes/di/RepositoryModule.java
package pan.alexander.notes.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import pan.alexander.notes.data.NotesRepositoryImplementation;
import pan.alexander.notes.data.TrashRepositoryImplementation;
import pan.alexander.notes.domain.NotesRepository;
import pan.alexander.notes.domain.TrashRepository;

@Module
public class RepositoryModule {

    @Provides
    @Singleton
    NotesRepository providesNotesRepository() {
        return new NotesRepositoryImplementation();
    }

    @Provides
    @Singleton
    TrashRepository providesTrashRepository() {
        return new TrashRepositoryImplementation();
    }
}

File: PopularLibs/fileconverter/app/src/main/java/pan/alexander/fileconverter/di/RepositoryModule.kt
package pan.alexander.fileconverter.di

import dagger.Binds
import dagger.Module
import pan.alexander.fileconverter.data.MainRepositoryImpl
import pan.alexander.fileconverter.domain.MainRepository

@Module
abstract class RepositoryModule {

    @Binds
    abstract fun provideMainRepository(mainRepository: MainRepositoryImpl): MainRepository
}
